account_invoice_production_lot/models/account_invoice.py

```diff
@@ -34,42 +24,21 @@ class AccountInvoiceLine(models.Model):
     @api.multi
     def _compute_prod_lots(self):
         for line in self:
-            line.prod_lot_ids = line.move_line_ids.mapped('lot_ids')
+            line.prod_lot_ids = line.mapped(
+                'move_line_ids.move_line_ids.lot_id'
+            )
 
     @api.multi
     def _compute_line_lots(self):
         for line in self:
-            if line.prod_lot_ids:
-                res = line.quantity_by_lot()
-                note = u'<ul>'
-                lot_strings = []
-                for lot in line.prod_lot_ids:
-                    lot_string = u'<li>S/N %s%s</li>' % (
-                        lot.name, u' (%s)' % res[lot] if res.get(lot) else '')
-                    lot_strings.append(lot_string)
-                note += u' '.join(lot_strings)
-                note += u'</ul>'
+            note = '<ul>'
+            lot_strings = []
+            for sml in line.mapped('move_line_ids.move_line_ids'):
+                if sml.lot_id:
+                    lot_strings.append('<li>%s %s (%s)</li>' % (
+                        _('S/N'), sml.lot_id.name, sml.qty_done,
```

Comment:
Shouldn't we check if the product is traced by serial number or lot to set this prefix accordingly?

Reply:
Well, that would be an improvement, yes. This is what was before...

Reply:
I have made the improvement to show the word and quantity in case of lots and S/N and no quantity with serials.